src/Draco.LanguageServer/Capabilities/TextDocumentSync.cs

```diff
@@ -17,8 +17,11 @@ public async Task TextDocumentDidOpenAsync(DidOpenTextDocumentParams param, Canc
         await this.PublishDiagnosticsAsync(uri);
     }
 
-    public Task TextDocumentDidCloseAsync(DidCloseTextDocumentParams param, CancellationToken cancellationToken) =>
-        Task.CompletedTask;
+    public async Task TextDocumentDidCloseAsync(DidCloseTextDocumentParams param, CancellationToken cancellationToken)
+    {
+        var uri = param.TextDocument.Uri;
+        await this.DeleteDocument(uri);
+    }
```

Comment:
Closing doesn't necessarily signal deletion, if I close a file, that doesn't mean that my still open files can't see stuff defined in it anymore.